storm-core/src/jvm/org/apache/storm/metrics2/StormMetricRegistry.java

```diff
@@ -0,0 +1,181 @@
+             LOG.warn("Unable to determine hostname while starting the metrics system. Hostname will be reported" +
+                     " as 'localhost'.");
+        }
+
+        LOG.info("Starting metrics reporters...");
+        List<Map<String, Object>> reporterList = (List<Map<String, Object>>)stormConfig.get(Config.STORM_METRICS_REPORTERS);
+        if(reporterList != null && reporterList.size() > 0) {
+            for (Map<String, Object> reporterConfig : reporterList) {
+                // only start those requested
+                List<String> daemons = (List<String>) reporterConfig.get("daemons");
+                for (String daemon : daemons) {
+                    if (DaemonType.valueOf(daemon.toUpperCase()) == type) {
+                        startReporter(stormConfig, reporterConfig);
+                    }
+                }
+            }
+        }
+    }
+
+    public static MetricRegistry registry(){
+        return REGISTRY;
+    }
+
+    private static void startReporter(Map<String, Object> stormConfig, Map<String, Object> reporterConfig){
+        String clazz = (String)reporterConfig.get("class");
+        StormReporter reporter = null;
+        LOG.info("Attempting to instantiate reporter class: {}", clazz);
+        reporter = Utils.newInstance(clazz);
+        if(reporter != null){
+            reporter.prepare(REGISTRY, stormConfig, reporterConfig);
+            reporter.start();
+            REPORTERS.add(reporter);
+        }
+
+    }
+
+    public static void stop(){
+        for(StormReporter sr : REPORTERS){
+            sr.stop();
+        }
+    }
+
+    public static String metricName(String name, String stormId, String componentId, String streamId, Integer taskId, Integer workerPort){
+        StringBuilder sb = new StringBuilder("storm.worker.");
+        sb.append(stormId);
+        sb.append(".");
+        sb.append(hostName);
+        sb.append(".");
+        sb.append(dotToUnderScore(componentId));
+        sb.append(".");
+        sb.append(dotToUnderScore(streamId));
+        sb.append(".");
+        sb.append(taskId);
+        sb.append(".");
+        sb.append(workerPort);
+        sb.append("-");
+        sb.append(name);
+        return sb.toString();
+    }
+
+    public static String metricName(String name, String stormId, String componentId, Integer workerPort) {
+        StringBuilder sb = new StringBuilder("storm.worker.");
+        sb.append(stormId);
+        sb.append(".");
+        sb.append(hostName);
+        sb.append(".");
+        sb.append(dotToUnderScore(componentId));
+        sb.append(".");
+        sb.append(workerPort);
+        sb.append("-");
+        sb.append(name);
+        return sb.toString();
+    }
+
+    public static String metricName(String name, TopologyContext context){
+        StringBuilder sb = new StringBuilder("storm.topology.");
+        sb.append(context.getStormId());
+        sb.append(".");
+        sb.append(hostName);
```

Comment:
hostname is expected to have '.' in it.   This makes parsing out any of the data after a host name close to impossible, this goes for all variants of the metric names.

Reply:
Are you saying you want '.' in hostname or not? Currently the hostname is reported with '.' replaced by '_'.  Leaving the '.' in hostname would make parsing anything after hostname impossible because a hostname contains an unpredictable number of dots.

Reply:
My bad I was doing some unit tests and I missed it in the other part of the code.  This is fine.